features/environment.py

```diff
@@ -65,8 +65,12 @@ def after_scenario(context, scenario):
     # we need to clean them up before behave starts appending new ones.
     
     if context.failed:
-        if not 'Behave errors' in context.step.error_message: #exclude behave output from exception logging
-            context.caught_exceptions.append(ExceptionSummary.from_context(context))
+        execution_mode = context.config.userdata.get('execution_mode')
+        if execution_mode and execution_mode == 'ExecutionMode.TESTING':
```

Comment:
It originally is, but the behave context stack outputs it as a string. 
```
context.config.userdata.get('execution_mode')
'ExecutionMode.TESTING'
type(context.config.userdata.get('execution_mode'))
<class 'str'>
```

What we can do is how we solved this earlier in the code, with using `eval`. Then it's again an enumeration. 
```
eval(context.config.userdata.get('execution_mode'))
<ExecutionMode.TESTING: 0>
type(eval(context.config.userdata.get('execution_mode')))
<enum 'ExecutionMode'>
eval(context.config.userdata.get('execution_mode')) == ExecutionMode.TESTING
True
```

Reply:
I also bumped into this from time to time. Just a general recommendation from my side not to use eval().

Reply:
Just out of curiosity, why? A reason I can come up with is that it would perhaps be safer to pass data around with the os environment rather than the behave stack. 
```
os.environ["EXECUTION_MODE"] = execution_mode.name
```

And retrieve it; 
```
execution_mode = ExecutionMode[os.environ.get('EXECUTION_MODE')]
execution_mode == ExecutionMode.TESTING
True
```